core/src/main/java/org/apache/mesos/offer/DefaultOfferRequirementProvider.java

```diff
@@ -155,4 +156,35 @@ private void validateVolumes(Protos.TaskInfo taskInfo, TaskSpecification taskSpe
 
         return volumes;
     }
+
+    /**
+     * Creates an {@link Protos.org.apache.mesos.Protos.ExecutorInfo} that drives
+     * the {@link org.apache.mesos.executor.CustomExecutor}.
+     * @param taskSpecification The {@link TaskSpecification} used to setup relevant environment for the executor.
+     * @return The {@link org.apache.mesos.Protos.ExecutorInfo} to run
+     * the {@link org.apache.mesos.executor.CustomExecutor}
+     * @throws NullPointerException
+     */
+    private Protos.ExecutorInfo getExecutorInfo(TaskSpecification taskSpecification) throws NullPointerException {
+
+        Protos.CommandInfo.URI executorURI;
+        Protos.ExecutorInfo.Builder executorInfoBuilder = Protos.ExecutorInfo.newBuilder()
+                .setName(taskSpecification.getName())
+                .setExecutorId(Protos.ExecutorID.newBuilder().setValue("").build()); // Set later by ExecutorRequirement
+
+        try {
+            executorURI = TaskUtils.uri(System.getenv(EXECUTOR_URI));
+        } catch (NullPointerException e) {
+            throw new NullPointerException(String.format("Failed at setting the URI for the executor: %s", e));
+        }
+
+        Protos.CommandInfo.Builder commandInfoBuilder = Protos.CommandInfo.newBuilder()
+                .setValue("./executor/bin/executor")
+                .addUris(executorURI)
+                .setEnvironment(taskSpecification.getCommand().getEnvironment())
```

Comment:
I don't have a `TaskInfo` or `ExecutorInfo` here though so not exactly sure what you mean. And based on my tests I didn't see the env vars being set by the CustomExecutor if I didn't explicitly set them here as part of the `ExecutorInfo`.